apps/src/blockly/addons/cdoXml.js

```diff
@@ -81,22 +81,39 @@ export default function initializeBlocklyXml(blocklyWrapper) {
 
     const positionBlock = function(block) {
       const heightWidth = block.blockly_block.getHeightWidth();
+      const hasFrameSvg = block.blockly_block.functionalSvg_ ? true : false;
+      const frameSvgSize = hasFrameSvg ? 40 : 0;
+      const frameSvgTop = hasFrameSvg ? 35 : 0;
+      const frameSvgMargin = hasFrameSvg ? 16 : 0;
 
       if (isNaN(block.x)) {
-        block.x = cursor.x;
+        block.x = blockSpace.RTL
+          ? cursor.x - frameSvgMargin
+          : cursor.x + frameSvgMargin;
       } else {
-        block.x = blockSpace.RTL ? viewWidth - block.x : block.x;
+        block.x = blockSpace.RTL
+          ? viewWidth - block.x - frameSvgMargin
+          : block.x + frameSvgMargin;
       }
```

Comment:
Nit: instead of ternarys in each of the if/else conditions, maybe just do this once after if/else:
```
blockSpace.RTL ?
  block.x -= frameSvgMargin :
  block.x += frameSvgMargin
```